File: app/src/main/java/com/ataulm/whatsnext/api/ApiContributor.java
package com.ataulm.whatsnext.api;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

class ApiContributor {

    @SerializedName("id")
    String id;

    @SerializedName("name")
    String name;

    @Nullable // only on actors
    @SerializedName("characterName")
    String characterName;
}

File: letterboxd/src/main/java/com/ataulm/letterboxd/ApiContributor.kt
package com.ataulm.letterboxd

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ApiContributor(
    @field:Json(name = "id")
    val id: String,

    @field:Json(name = "name")
    val name: String,

    /**
     * Only available for actors
     */
    @field:Json(name = "characterName")
    val characterName: String?
)
